src/runtimes/node/native_modules/plugin.js

```diff
@@ -0,0 +1,68 @@
+const path = require('path')
+
+const readPackageJson = require('read-package-json-fast')
+
+const { isNativeModule } = require('./detector')
+
+const packageName = /^([^@][^/]*|@[^/]*\/[^/]+)(?:\/|$)/
+
+const findNativeModule = (packageJsonPath, cache) => {
+  if (cache[packageJsonPath] === undefined) {
+    // eslint-disable-next-line fp/no-mutation, no-param-reassign, promise/prefer-await-to-then
+    cache[packageJsonPath] = readPackageJson(packageJsonPath).then(
+      (data) => Boolean(isNativeModule(data)),
+      () => {},
+    )
+  }
+
+  return cache[packageJsonPath]
+}
+
+const externalNativeModulesPlugin = (externalizedModules) => ({
+  name: 'external-native-modules',
+  setup(build) {
+    const cache = {}
+
+    // eslint-disable-next-line complexity, max-statements
+    build.onResolve({ filter: /^([^./]*)$/ }, async (args) => {
```

Comment:
[sand] can we extract the regex to a const with a meaningful name so it is clear what we're filtering?

Reply:
Absolutely!

Reply:
Done in https://github.com/netlify/zip-it-and-ship-it/pull/418/commits/f47f22206c4e80d9e921b7b1e2af96d8dc95d379.